overlord/ifacestate/helpers.go

```diff
@@ -163,6 +165,9 @@ func (m *InterfaceManager) regenerateAllSecurityProfiles() error {
 		}
 	}
 
+	writeSystemKey := true
+	os.Remove(dirs.SnapSystemKeyFile)
```

Comment:
This should not be needed - we only end up in this code path if `profilesNeedRegeneration()` detected a systemKey mismatch. So the change to only write the systemKey if the profile was successful should be sufficient (this is the only place in the code that writes the system key). Or am I missing something?

Reply:
Did you read the commit message of the patch adding this?

Reply:
Thanks, seems worthwhile to add a comment about this into the code though (copy/paste of the relevant bits from the commit message should be fine).